include/xtensor/xcontainer.hpp

```diff
@@ -358,8 +361,57 @@ namespace xt
         adapt_strides();
         data().resize(data_size());
     }
+
+    /**
+     * Transposes the container inplace by reversing the dimensions.
+     */
+    template <class D>
+    inline void xcontainer<D>::transpose()
+    {
+        // reverse stride and shape
+        std::reverse(m_strides.begin(), m_strides.end());
+        std::reverse(m_shape.begin(), m_shape.end());
+        adapt_strides();
+    }
+
+    /**
+     * Transposes the container inplace by permuting the shape with @permute.
+     * @param permute the dimensions with the given permutation
+     */
+    template <class D>
+    inline void xcontainer<D>::transpose(const std::vector<size_type>& permute)
+    {
+        if (permute.size() != m_shape.size())
+            throw transpose_error(std::string("Permutation does not have the same size as shape"));
```

Comment:
This check is too heavy to be the default behavior.  A better solution would be to have an additional parameter (let's call it check_policy) with a default value equal to no_check. This parameter could be a simple tag and the implementation of the `transpose` function would call a private `check_permutation` function (which overload to call would be resolved by tag dispatching).